src/ctrl.c

```diff
@@ -187,8 +187,10 @@ static gboolean read_from_ctrl_buffer(int fd, gboolean (*line_process_func)(char
 	char *newline = strchrnul(beg, '\n');
 	/* Process each message which ends with a line */
 	while (*newline != '\0') {
-		if (!line_process_func(ctlbuf))
+		if (!line_process_func(beg))
 			return G_SOURCE_CONTINUE;
+		// Restore the newline character
+		*newline = '\n';
```

Comment:
hmm where did it go?

Reply:
I that was an overkill. The newline character isn't needed anymore - once we've processed a line and null-terminated it, we don't need to restore the newline since we're done with that part of the buffer.

Reply:
Also, tested this change and it works.

Reply:
@haircommander PTAL